version/check.go

```diff
@@ -0,0 +1,64 @@
+package version
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+)
+
+const versionTimeout = time.Second * 1
+
+// CheckLatestVersion prints the latest version if the current version is less than latest version
+func CheckLatestVersion(currVersion string, githubReleaseURL string) error {
+	gitClient := http.Client{
+		Timeout: versionTimeout,
+	}
+
+	req, err := http.NewRequest(http.MethodGet, githubReleaseURL, nil)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create request for latest version")
+	}
+	req.Header.Set("User-Agent", "salt")
+	res, err := gitClient.Do(req)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get latest version from github")
+	}
+	if res.StatusCode != http.StatusOK {
+		return errors.Wrapf(err, "failed to get latest version from github")
+	}
+	if res.Body != nil {
+		defer res.Body.Close()
+	}
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read response body")
+	}
+
+	authorType := struct {
+		TagName string `json:"tag_name"`
+	}{}
+	if err = json.Unmarshal(body, &authorType); err != nil {
+		return errors.Wrapf(err, "failed to parse: %s", string(body))
+	}
+
+	currentV, err := version.NewVersion(currVersion)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse current version")
+	}
+	fmt.Sprintf("current version: %s", currVersion)
```

Comment:
This is not actually printing. `fmt.Sprintf` just returns the formatted string.

Reply:
one more such occurrence

Reply:
Yes, we should just return a string from the main function